shared/kubernetes/k8sTapManager.go

```diff
@@ -0,0 +1,217 @@
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+	"github.com/up9inc/mizu/shared"
+	"github.com/up9inc/mizu/shared/debounce"
+	"github.com/up9inc/mizu/shared/goUtils"
+	"github.com/up9inc/mizu/shared/logger"
+	"github.com/up9inc/mizu/tap/api"
+	core "k8s.io/api/core/v1"
+	"regexp"
+	"time"
+)
+
+const updateTappersDelay = 5 * time.Second
+
+type TappedPodChangeEvent struct {
+	Added   []core.Pod
+	Removed []core.Pod
+}
+
+// MizuTapperSyncer uses a k8s pod watch to update tapper daemonsets when targeted pods are removed or created
+type MizuTapperSyncer struct {
+	context             context.Context
+	CurrentlyTappedPods []core.Pod
+	config              TapperSyncerConfig
+	kubernetesProvider  *Provider
+	TapPodChangesOut    chan TappedPodChangeEvent
+	ErrorOut            chan K8sTapManagerError
+	shouldUpdateTappers bool // Used to prevent daemonset updates but still allow tracking targeted pods
+}
+
+type TapperSyncerConfig struct {
+	TargetNamespaces         []string
+	PodFilterRegex           regexp.Regexp
+	MizuResourcesNamespace   string
+	AgentImage               string
+	TapperResources          shared.Resources
+	ImagePullPolicy          core.PullPolicy
+	DumpLogs                 bool
+	IgnoredUserAgents        []string
+	MizuApiFilteringOptions  api.TrafficFilteringOptions
+	MizuServiceAccountExists bool
+}
+
+func CreateAndStartMizuTapperSyncer(ctx context.Context, kubernetesProvider *Provider, config TapperSyncerConfig, shouldUpdateTappers bool) (*MizuTapperSyncer, error) {
+	manager := &MizuTapperSyncer{
+		context:             ctx,
+		CurrentlyTappedPods: make([]core.Pod, 0),
+		config:              config,
+		kubernetesProvider:  kubernetesProvider,
+		TapPodChangesOut:    make(chan TappedPodChangeEvent, 100),
+		ErrorOut:            make(chan K8sTapManagerError, 100),
+		shouldUpdateTappers: shouldUpdateTappers,
```

Comment:
in cli you can run k8s get pods machining regex, print to user 
you can use list of currently tapped pods